### PR TITLE
add github action for tests

### PR DESCRIPTION
**Pull request by wpoely86**
It fails at this moment: https://github.com/wpoely86/ncm-ncd/actions/runs/3592724123

### CODE REVIEW

**Comment by wpoely86** [comment]
I will have a look but it is running as root at this moment.

**Comment by wpoely86** [comment]
I'm managed to fix one of the tests. The other one that fails is:

```
# 2 failed comps
$mock_cpl->mock('_mtime', sub {return $_[0] =~ m/woohaa/ ? 123456789 : 987654321});
set_state($this_app, 'woohaa', 'woopsie', $statedir);
set_state($this_app, 'ouch', '', $statedir);
is_deeply(get_states($this_app, $statedir), {
    woohaa => {
        message => "woopsie\n",
        timestamp => 123456789,
    },
    ouch => {
        message => "\n",
        timestamp => 987654321,
    }
}, "get_states on 2 failed components returns hashref");

@pri = qw();
eval {$this_app->main($ec);};
like($@, qr{^exit -1 at}, "report exited with failure");
ok($this_app->directory_exists($statedir), "statedir $statedir does exist 3");
diag explain \@pri;
is($pri[-3]->[0], '2 components with error', "Main report is 2 failed components");
like($pri[-2]->[0], qr{^  ouch failed on .*? 2001 \(no message\)}, "failed ouch component");
like($pri[-1]->[0], qr{^  woohaa failed on .*? 1973 with message woopsie}, "failed woohaa component");
```

It doesn't find the 'ouch failed' output. No clue what is the matter with this test.

**Comment by jrha** [comment]
I just realised that this is the same problem that has been breaking nightlies on EL6 and EL7 for a while.
It works fine on EL8 though.

File: src/test/perl/tabcompletion.t [hunk 1]
```diff
@@ -25,12 +25,12 @@ $mock_cli->mock('_get_uid', 0);
 # Do not mock the renderer, use the one from CCM
 # try source relative to current path (eg on jenkins) and default install
 my $tt_ccm = abs_path(getcwd().'/../CCM/src/main/resources');
-ok(-d $tt_ccm, "Did not find the CCM TT directory $tt_ccm");
+ok(-d $tt_ccm, "Did not find the CCM TT directory $tt_ccm") if defined $tt_ccm;
```
Comment (jrha) [inline src/test/perl/tabcompletion.t:28]: Ah, so the tests need CCM to be checked out somewhere nearby?

Comment (stdweird) [reply src/test/perl/tabcompletion.t:28]: during the initial setup, you actually install the CCM rpm, so you should fallback to the rpm path before giving up

**Comment by jrha** [comment]
@wpoely86 can you change this one over to `rockylinux:8` as well?

**Comment by wpoely86** [comment]
> @wpoely86 can you change this one over to `rockylinux:8` as well?

Done, output at https://github.com/wpoely86/ncm-ncd/actions/

**Comment by jrha** [comment]
Awesome thanks :smile:

**Review by jrha — APPROVED** [review]

**Pull request merged**